src/mesh/voxel_cloud.rs

```diff
@@ -128,111 +128,261 @@ impl VoxelCloud {
     /// (relative to the voxel space origin).
+    #[allow(dead_code)]
     pub fn voxel_at_absolute_coords(&self, absolute_coords: &Point3<i32>) -> Option<bool> {
-        self.absolute_three_dimensional_coordinate_to_one_dimensional(absolute_coords)
-            .map(|index| self.voxel_map[index])
+        absolute_three_dimensional_coordinate_to_one_dimensional(
+            absolute_coords,
+            &self.block_start,
+            &self.block_dimensions,
+        )
+        .map(|index| self.voxel_map[index])
     }
 
     /// Gets the state of a voxel containing the input point defined in model
     /// space coordinates.
     #[allow(dead_code)]
     pub fn voxel_at_cartesian_coords(&self, point: &Point3<f32>) -> Option<bool> {
-        let voxel_coords = self.cartesian_to_absolute_voxel_coords(point);
+        let voxel_coords = cartesian_to_absolute_voxel_coords(point, &self.voxel_dimensions);
         self.voxel_at_absolute_coords(&voxel_coords)
     }
 
     /// Sets the state of a voxel defined in voxel coordinates relative to the
     /// voxel block start.
+    #[allow(dead_code)]
     pub fn set_voxel_at_relative_coords(&mut self, relative_coords: &Point3<i32>, state: bool) {
-        let index = self
-            .relative_three_dimensional_coordinate_to_one_dimensional(relative_coords)
-            .expect("Coordinates out of bounds");
+        let index = relative_three_dimensional_coordinate_to_one_dimensional(
+            relative_coords,
+            &self.block_dimensions,
+        )
+        .expect("Coordinates out of bounds");
         self.voxel_map[cast_usize(index)] = state;
     }
 
     /// Sets the state of a voxel defined in absolute voxel coordinates
     /// (relative to the voxel space origin).
     pub fn set_voxel_at_absolute_coords(&mut self, absolute_coords: &Point3<i32>, state: bool) {
-        let index = self
-            .absolute_three_dimensional_coordinate_to_one_dimensional(absolute_coords)
-            .expect("Coordinates out of bounds");
+        let index = absolute_three_dimensional_coordinate_to_one_dimensional(
+            absolute_coords,
+            &self.block_start,
+            &self.block_dimensions,
+        )
+        .expect("Coordinates out of bounds");
         self.voxel_map[cast_usize(index)] = state;
     }
 
     /// Sets the state of a voxel containing the input point defined in model
     /// space coordinates.
     pub fn set_voxel_at_cartesian_coords(&mut self, point: &Point3<f32>, state: bool) {
-        let voxel_coords = self.cartesian_to_absolute_voxel_coords(point);
+        let voxel_coords = cartesian_to_absolute_voxel_coords(point, &self.voxel_dimensions);
         self.set_voxel_at_absolute_coords(&voxel_coords, state);
     }
 
+    /// Resize the voxel cloud block to match new block start and block dimensions.
+    ///
+    /// This clips the outstanding parts of the original voxel cloud.
+    #[allow(dead_code)]
```

Comment:
`#[allow(dead_code)]` works recursively. Only the following really need to be marked:

- `block_start`
- `block_dimensions`
- `voxel_dimensions`
- `voxel_map_len`
- `voxel_at_cartesian_coords`
- `set_voxel_at_relative_coords`
- `shrink_to_fit`
- `cartesian_to_relative_voxel_coords`
- `absolute_voxel_to_cartesian_coords`

(This already includes that some of the trivial dead functions shouldn't probably be included)